File: Day 78/Solution.py
# 520. Detect Capital
# We define the usage of capitals in a word to be right when one of the following cases holds:

# All letters in this word are capitals, like "USA".
# All letters in this word are not capitals, like "leetcode".
# Only the first letter in this word is capital, like "Google".
# Given a string word, return true if the usage of capitals in it is right.

class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        count = 0
        for ch in word:
            if ch.isupper():
                count += 1

        if count == len(word) or count == 0:
            return True
        if count == 1 and word[0].isupper():
            return True
        return False

if __name__ == "__main__":
    sol = Solution()
    print(sol.detectCapitalUse("USA"))      # True
    print(sol.detectCapitalUse("leetcode")) # True
    print(sol.detectCapitalUse("Google"))   # True
    print(sol.detectCapitalUse("FlaG"))     # False

File: Day 78/Solution.java
// 520. Detect Capital
// We define the usage of capitals in a word to be right when one of the following cases holds:

// All letters in this word are capitals, like "USA".
// All letters in this word are not capitals, like "leetcode".
// Only the first letter in this word is capital, like "Google".
// Given a string word, return true if the usage of capitals in it is right.
class Solution {
    public boolean detectCapitalUse(String word) {
        int count = 0;
        for (char ch : word.toCharArray()) {
            if (Character.isUpperCase(ch)) {
                count++;
            }
        }

        if (count == word.length() || count == 0) {
            return true;
        }
        if (count == 1 && Character.isUpperCase(word.charAt(0))) {
            return true;
        }
        return false;
    }

    // For VS Code testing
    public static void main(String[] args) {
        Solution sol = new Solution();
        System.out.println(sol.detectCapitalUse("USA"));     // true
        System.out.println(sol.detectCapitalUse("leetcode")); // true
        System.out.println(sol.detectCapitalUse("Google"));   // true
        System.out.println(sol.detectCapitalUse("FlaG"));     // false
    }
}
